backends/dpdk/dpdkAsmOpt.cpp

```diff
@@ -163,4 +163,32 @@ const IR::IndexedVector<IR::DpdkAsmStatement> *RemoveLabelAfterLabel::removeLabe
     }
     return new_l;
 }
+
+const IR::Node* RemoveUnusedMetadataFields::preorder(IR::DpdkAsmProgram *p) {
+    auto usedFieldsCollector = new CollectUsedMetadataField();
+    usedFieldsCollector->setCalledBy(this);
+    p->apply(*usedFieldsCollector);
+    auto used_fields = usedFieldsCollector->getUsedFields();
+
+    IR::IndexedVector<IR::DpdkStructType> usedStruct;
+    for (auto st : p->structType) {
+        IR::IndexedVector<IR::StructField> usedMetadataFields;
+        for (auto field : st->fields) {
+            if (used_fields.count("m." + field->name.name)) {
```

Comment:
While storing the fields names in used_fields, "m." can be truncated. So this check would become simpler.

Reply:
done